docs/reference/networking.md

```diff
@@ -1,13 +1,89 @@
+hardware MAC addresses and optionanlly configured by plugin [VLAN](https://en.wikipedia.org/wiki/Virtual_LAN).
+
 # Supported CNI implementations
 
+Virtlet was tested and is working correctly with:
+
+* [Flannel](https://github.com/coreos/flannel)
+* [Calico](https://github.com/projectcalico/cni-plugin)
+* [Weave](https://github.com/weaveworks/weave)
+* [original](https://github.com/hustcat/sriov-cni) and [Intel fork](https://github.com/intel/sriov-cni) of SR-IOV
+* [combination](#multi-cni) of them with [CNI-Genie](https://github.com/Huawei-PaaS/CNI-Genie)
+
 # Configuring using DHCP
 
+Information collected from CNI plugin is used by DHCP server accessible on each
+of configured for VM interfaces. That data is visible only for particular VM
+and is limited to serve configuration for particular interface, per interface.
+
+DHCP configuration is unavailable for [SR-IOV](#sr-iov) devices as they are
+passed directly to VM while fully excluded from host kernel configuration.
+
 # Configuring using Cloud-Init
 
-Not supported for persistent rootfs for now
+In addition to passing network configuration through DHCP it is also passed to
+VM as a part of [Cloud-init](./cloud-init.md) data. That is the preffered
+source of configuration data for SR-IOV devices.
+Data configured by Virtlet is formed following
+[Network Config Version 1](https://cloudinit.readthedocs.io/en/latest/topics/network-config-format-v1.html)
+format.
+
+Note: Cloud-init network configuration is not supported for persistent rootfs
+for now.
 
-# Setting up Multiple CNIs
+# <a name="multi-cni"></a> Setting up Multiple CNIs
+
+Virtlet allows to configure multiple interfaces for VM when all of them are
+properly described in
+[CNI Result](https://github.com/containernetworking/cni/blob/spec-v0.3.1/SPEC.md#result).
+The supported way to achieve that using CNI plugins is by combining
+output of their chain using [CNI-Genie](https://github.com/Huawei-PaaS/CNI-Genie).
+
+Before you proceed, please read the CNI Genie [documentation](https://github.com/Huawei-PaaS/CNI-Genie/blob/master/docs/CNIGenieFeatureSet.md).
+There are two ways to tell CNI Genie which CNI networks to use:
+
+* by setting `cni: <plugins list>` in pods annotation
+* by setting the `default_plugin` option in CNI Genie configuration file
+
+Note: when you are planning to use Calico plugin - it needs to be used as first
+in chain, or other plugins prior to it need to leave default route not
+configured. That is specifical limitation of Calico plugin.
+
+## Configuring networks
+
+CNI plugins should return result in at least 0.3.0 format. All config files should be updated.
+
+## Example configuration
+
+Please reffer to [more detailed documentation](https://github.com/Mirantis/virtlet/blob/master/docs/multiple-interfaces.md#example-files)
+in this topic for example configuration files using CNI Genie with Calico for
+primary and Flannel for secondary interface.
```

Comment:
Please refer to the [detailed documentation](https://github.com/Mirantis/virtlet/blob/master/docs/multiple-interfaces.md#example-files)
that contains an example of configuration files for CNI Genie with Calico being used for
the primary interface and Flannel being used for the secondary one.